### DISCUSSION TITLE
Fedora - Native Chrome

### DISCUSSION THREAD

**Discussion by repires** [aclap-dev/video-downloadhelper#404] [Help and Q&A] [answered]
### Your question

Hi, I'm not able to use it on my system. Firefox native works fine.
Download link: curl -sSLf https://github.com/aclap-dev/vdhcoapp/releases/latest/download/install.sh | bash

/opt/vdhcoapp/vdhcoapp --info
zsh: no such file or directory: /opt/vdhcoapp/vdhcoapp

Checking companion app returned: Specified native messaging host not found.. [Click here to troubleshoot your issue.](https://github.com/aclap-dev/video-downloadhelper/wiki/CoApp-not-recognized)

This tutorial doesn't work: https://discussion.fedoraproject.org/t/how-to-install-vdhcoapp-needed-to-use-downloadhelper-addones/77149/12

This copr package also does not solve the problem: https://discussion.fedoraproject.org/t/rok-vdhcoapp/108363

# System Details Report
---

## Report details
- **Date generated:**                              2024-04-15 13:39:21

## Hardware Information:
- **Hardware Model:**                              Dell Inc. Inspiron 3583
- **Memory:**                                      8.0 GiB
- **Processor:**                                   Intel® Core™ i7-8565U × 8
- **Graphics:**                                    Intel® UHD Graphics 620 (WHL GT2)
- **Graphics 1:**                                  HAINAN
- **Disk Capacity:**                               (null)

## Software Information:
- **Firmware Version:**                            1.29.0
- **OS Name:**                                     Fedora Linux 39 (Workstation Edition)
- **OS Build:**                                    (null)
- **OS Type:**                                     64-bit
- **GNOME Version:**                               45.5
- **Windowing System:**                            Wayland
- **Kernel Version:**                              Linux 6.8.5-201.fc39.x86_64

Addon & CoApp version that's working in Firefox rpm native app
version: 8.2.2.8
channel: stable
build date: 2024-03-18
build options: linuxlic: false, noyt: false, browser: firefox
lang: en-US
license: unneeded
platform: x86-64 linux
UA: Mozilla/5.0 (X11; Linux x86_64; rv:124.0) Gecko/20100101 Firefox/124.0
coapp: yes. Info:
{
  "id": "net.downloadhelper.coapp",
  "name": "VdhCoApp",
  "version": "2.0.19",
  "binary": "/home/rnt/.local/share/vdhcoapp/vdhcoapp",
  "displayName": "VdhCoApp",
  "description": "Video DownloadHelper companion app",
  "target": {
    "os": "linux",
    "arch": "x86_64",
    "node": "18"
  },
  "home": "/home/rnt",
  "converterBinary": "/home/rnt/.local/share/vdhcoapp/ffmpeg",
  "converterBase": "ffmpeg",
  "converterBaseVersion": "c30f360"
}

### Addon & CoApp version and details

version: 8.2.2.9
channel: stable
build date: 2024-03-19
build options: linuxlic: false, noyt: true, browser: chrome
lang: en-GB
license: unneeded
platform: x86-64 linux
UA: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/123.0.0.0 Safari/537.36
coapp: no.

**Comment by paulrouget** [comment]
Is it Chrome Flatpak?

**Comment by repires** [comment]
No, native .rpm fedora package

**Comment by paulrouget** [comment]
So there's no sandboxing involved whatsover?

What do these say?
```
/home/rnt/.local/share/vdhcoapp/vdhcoapp install
/home/rnt/.local/share/vdhcoapp/vdhcoapp --info
```

**Reply by repires** [reply]
Firefox and Chrome are installed as native .rpm Fedora package. Don't know why it's in this folder or why chrome doesn't recognize the coapp.

**Reply by paulrouget** [reply, marked as answer]
Please tell me what these say:

```
/home/rnt/.local/share/vdhcoapp/vdhcoapp install
/home/rnt/.local/share/vdhcoapp/vdhcoapp --info
```

**Reply by paulrouget** [reply]
These are commands to run.

**Reply by repires** [reply]
Sorry, it was my mistake, indeed it's the flatpak version, my wife changed. 
Now it worked, thanks! 
Does it has to do anything about Fedora system install?

```
/home/rnt/.local/share/vdhcoapp/vdhcoapp install

Installing…
Flatpak is installed. Making the coapp available from browser sandboxes:
Linked coapp within org.mozilla.firefox.
Linked coapp within com.brave.Browser.
Linked coapp within com.google.Chrome.
Linked coapp within com.google.ChromeDev.
Linked coapp within org.chromium.Chromium.
Linked coapp within com.github.Eloston.UngoogledChromium.
Linked coapp within com.microsoft.Edge.
Linked coapp within com.microsoft.EdgeDev.
Writing /home/rnt/.mozilla/native-messaging-hosts/net.downloadhelper.coapp.json
Writing /home/rnt/.var/app/org.mozilla.firefox/.mozilla/native-messaging-hosts/net.downloadhelper.coapp.json
Writing /home/rnt/.config/google-chrome/NativeMessagingHosts/net.downloadhelper.coapp.json
Writing /home/rnt/.config/chromium/NativeMessagingHosts/net.downloadhelper.coapp.json
Writing /home/rnt/.var/app/com.brave.Browser/config/BraveSoftware/Brave-Browser/NativeMessagingHosts/net.downloadhelper.coapp.json
Writing /home/rnt/.var/app/com.google.Chrome/config/google-chrome/NativeMessagingHosts/net.downloadhelper.coapp.json
Writing /home/rnt/.var/app/com.github.Eloston.UngoogledChromium/config/chromium/NativeMessagingHosts/net.downloadhelper.coapp.json
VdhCoApp : VdhCoApp is ready to be used
```

```
/home/rnt/.local/share/vdhcoapp/vdhcoapp --info

{
  "id": "net.downloadhelper.coapp",
  "name": "VdhCoApp",
  "version": "2.0.19",
  "binary": "/home/rnt/.local/share/vdhcoapp/vdhcoapp",
  "displayName": "VdhCoApp",
  "description": "Video DownloadHelper companion app",
  "target": {
    "os": "linux",
    "arch": "x86_64",
    "node": "18"
  },
  "home": "/home/rnt",
  "converterBinary": "/home/rnt/.local/share/vdhcoapp/ffmpeg",
  "converterBase": "ffmpeg",
  "converterBaseVersion": "c30f360"
}```
